gossip/src/duplicate_shred_handler.rs

```diff
@@ -131,6 +160,18 @@ impl DuplicateShredHandler {
                     shred1.into_payload(),
                     shred2.into_payload(),
                 )?;
+                // feature_epoch could only be 0 in tests and new cluster setup.
+                if self
+                    .enable_gossip_duplicate_proof_ingestion_epoch
+                    .is_some_and(|feature_epoch| {
+                        self.epoch_schedule.get_epoch(slot) > feature_epoch || feature_epoch == 0
+                    })
+                {
```

Comment:
Because of the blockstore insert check, this is already rate limited to at most once per slot.
So I think it would be easier to just check for feature activation here instead of storing `enable_gossip_duplicate_proof_ingestion_epoch` in a field and manually updating it.